pandas/tseries/resample.py

```diff
@@ -1189,13 +1190,18 @@ def _get_time_delta_bins(self, ax):
             raise TypeError('axis must be a TimedeltaIndex, but got '
                             'an instance of %r' % type(ax).__name__)
 
+        if len(ax) > 0 and all(ax._isnan):
```

Comment:
I would give a better error message here. Is this consistent with how we handle all-nan grouping?